ports/esp32/boards/GENERIC_S2/mpconfigboard.cmake

```diff
@@ -4,3 +4,25 @@ set(SDKCONFIG_DEFAULTS
     boards/sdkconfig.base
     boards/sdkconfig.usb
 )
+
+if(MICROPY_BOARD_VARIANT STREQUAL "wrover")
+    set(SDKCONFIG_DEFAULTS
+        ${SDKCONFIG_DEFAULTS}
+        boards/sdkconfig.spiram_sx
+        boards/GENERIC_S2/sdkconfig.wrover
+    )
+
+    list(APPEND MICROPY_DEF_BOARD
+        MICROPY_HW_I2C0_SCL=7
+        MICROPY_HW_I2C0_SDA=6
```

Comment:
Did you consider putting settings like this in `mpconfigboard.h`?  Eg:
```c
#if MICROPY_BOARD_VARIANT_WROVER
#define MICROPY_HW_I2C0_SCL 7
#endif
```
Probably it's cleaner to keep all variant config in the .cmake file (and .mk for make-based ports)...

Reply:
I did but there was no existing precedent for having the variant as a preprocessor macro, and exactly like you say it's neater putting it in one place.

Reply:
I'm pretty confused by this "wrover" variant anyway and why it needs to define the pins?

Reply:
Maybe because USB uses GPIO19 which is the default for I2C0 SDA?  But that would be a problem for all S2 boards.

Reply:
How do you feel about replacing this variant with a "spiram" variant, which is consistent with what we've done for OG and S3?

Reply:
I've pushed the branch with an extra commit that does this rename/replace. See the commit description for notes.